pkg/aggregator/registry/dataplaneservice/strategy.go

```diff
@@ -0,0 +1,165 @@
+	return false
+}
+
+func (dataPlaneServiceStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (dataPlaneServiceStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return field.ErrorList{}
+}
+
+// WarningsOnUpdate returns warnings for the given update.
+func (dataPlaneServiceStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
+	return nil
+}
+
+type dataPlaneServiceStatusStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// NewStatusStrategy creates a new dataPlaneServiceStatusStrategy.
+func NewStatusStrategy(typer runtime.ObjectTyper) rest.UpdateResetFieldsStrategy {
+	return dataPlaneServiceStatusStrategy{typer, names.SimpleNameGenerator}
+}
+
+func (dataPlaneServiceStatusStrategy) NamespaceScoped() bool {
+	return false
+}
+
+func (dataPlaneServiceStatusStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
+	fields := map[fieldpath.APIVersion]*fieldpath.Set{
+		"aggregation.grafana.app/v0alpha1": fieldpath.NewSet(
+			fieldpath.MakePathOrDie("spec"),
+			fieldpath.MakePathOrDie("metadata"),
+		),
+	}
+
+	return fields
+}
+
+func (dataPlaneServiceStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newAPIService := obj.(*aggregation.DataPlaneService)
+	oldAPIService := old.(*aggregation.DataPlaneService)
+	newAPIService.Spec = oldAPIService.Spec
+	newAPIService.Labels = oldAPIService.Labels
+	newAPIService.Annotations = oldAPIService.Annotations
+	newAPIService.Finalizers = oldAPIService.Finalizers
+	newAPIService.OwnerReferences = oldAPIService.OwnerReferences
+}
+
+func (dataPlaneServiceStatusStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (dataPlaneServiceStatusStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// Canonicalize normalizes the object after validation.
+func (dataPlaneServiceStatusStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// ValidateUpdate validates an update of dataPlaneServiceStatusStrategy.
+func (dataPlaneServiceStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return field.ErrorList{}
+}
+
+// WarningsOnUpdate returns warnings for the given update.
+func (dataPlaneServiceStatusStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
+	return nil
+}
+
+// GetAttrs returns the labels and fields of an API server for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	s, ok := obj.(*aggregation.DataPlaneService)
+	if !ok {
+		return nil, nil, fmt.Errorf("given object is not a DataPlaneService")
+	}
+	return labels.Set(s.ObjectMeta.Labels), ToSelectableFields(s), nil
+}
+
+// MatchAPIService is the filter used by the generic etcd backend to watch events
+// from etcd to clients of the apiserver only interested in specific labels/fields.
+func MatchAPIService(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
```

Comment:
Is MatchAPIService the correct name for this helper since we are calling the Kind a DataPlaneService elsewhere?

Reply:
updated in [a7c3e6d](https://github.com/grafana/grafana/pull/91228/commits/a7c3e6d4839c279e81a305ad8c26dd0d9b8e9bee) and added SPDX headers to point to the upstream source